src/shared/logs-show.c

```diff
@@ -333,8 +334,40 @@ static int output_timestamp_monotonic(FILE *f, sd_journal *j, const char *monoto
         if (r < 0)
                 return log_error_errno(r, "Failed to get monotonic timestamp: %m");
 
-        fprintf(f, "[%5"PRI_USEC".%06"PRI_USEC"]", t / USEC_PER_SEC, t % USEC_PER_SEC);
-        return 1 + 5 + 1 + 6 + 1;
+        snprintf(buf, sizeof(buf), "[%5"PRI_USEC".%06"PRI_USEC"", t / USEC_PER_SEC, t % USEC_PER_SEC);
+
+        if (mode == OUTPUT_SHORT_DELTA) {
+                sd_id128_t previous_boot_id;
+                uint64_t previous_t, delta;
+
+                r = sd_journal_previous(j);
```

Comment:
hmm, please don't do it like this. The last printed message and the previous message in the log stream might be two different things, and that would get quite confusing.

(we do pcre filtering after all, client side).

Hence, just add a new parameter to the output function, with the previous timestamp and boot id, and then just keep track of that in the invocating loop, do not try to reconstruct the right value here...

Reply:
Refactored. I hope it is what you expected it to be.